src/utilities/urlAssembly.spec.ts

```diff
@@ -1,83 +1,170 @@
+    const route = {
+      name: 'simple',
+      path,
+      component,
+    } satisfies Route
+    const [resolved] = resolveRoutes([route])
 
-  const url = assembleUrl(resolved, { simple: ['ABC'] })
+    const url = assembleUrl(resolved, { simple: ['ABC'] })
 
-  expect(url).toBe('/simple/ABC')
-})
+    expect(url).toBe('/simple/ABC')
+  })
+
+  test.each([
+    ['/simple/:simple'],
+    [path('/simple/:simple', { simple: String })],
+  ])('given route with required string param NOT provided, throws error', (path) => {
+    const route = {
+      name: 'simple',
+      path,
+      component,
+    } satisfies Route
+    const [resolved] = resolveRoutes([route])
+
+    expect(() => assembleUrl(resolved, {})).toThrowError(InvalidRouteParamValueError)
+  })
 
-test.each([
-  ['/simple/:simple'],
-  [path('/simple/:simple', { simple: String })],
-])('given route with required string param NOT provided, throws error', (path) => {
-  const route = {
-    name: 'simple',
-    path,
-    component,
-  } satisfies Route
-  const [resolved] = resolveRoutes([route])
-
-  expect(() => assembleUrl(resolved, {})).toThrowError(InvalidRouteParamValueError)
+  test.each([
+    ['/simple/:simple'],
+    [path('/simple/:simple', { simple: String })],
+  ])('given route with required string param provided, returns route Path with string with values interpolated', (path) => {
+    const route = {
+      name: 'simple',
+      path,
+      component,
+    } satisfies Route
+    const [resolved] = resolveRoutes([route])
+
+    const url = assembleUrl(resolved, { simple: ['ABC'] })
+
+    expect(url).toBe('/simple/ABC')
+  })
 })
 
-test.each([
-  ['/simple/:simple'],
-  [path('/simple/:simple', { simple: String })],
-])('given route with required string param provided, returns route Path with string with values interpolated', (path) => {
-  const route = {
-    name: 'simple',
-    path,
-    component,
-  } satisfies Route
-  const [resolved] = resolveRoutes([route])
+describe('query params', () => {
+  test.each([
+    ['simple=abc'],
+    [query('simple=abc', {})],
+  ])('given simple route with string query and without params, returns route query', (query) => {
+    const route = {
+      name: 'simple',
+      path: '/',
+      query,
+      component,
+    } satisfies Route
+    const [resolved] = resolveRoutes([route])
+
+    const url = assembleUrl(resolved, {})
+
+    expect(url).toBe('/?simple=abc')
+  })
+
+  test.each([
+    ['simple=:?simple'],
+    [query('simple=:?simple', { simple: String })],
+  ])('given route with optional string param NOT provided, returns route Query with string without values interpolated', (query) => {
+    const route = {
+      name: 'simple',
+      path: '/',
+      query,
+      component,
+    } satisfies Route
+    const [resolved] = resolveRoutes([route])
+
+    const url = assembleUrl(resolved, {})
+
+    expect(url).toBe('/?simple=')
```

Comment:
I'm not sure what the expected behavior here would be. This test makes sense for sure. But I wonder if users will expect the key to not be present when the value is absent. But I honestly like how simple this is.

Reply:
agreed. And unless you felt strongly I was going to just go with whatever is the most cut and dry and let our users tell us what improvements they want to see here